### PR TITLE
 vec:make vec lists dynamic

### PR DESCRIPTION
**Pull request by KeithWiles**
Make the vec lists more dynamic instead of static sizes. Part of this change
is because of the CNET/TCP update comming, but also makes the code simpler.

With this update the old vec_*_ptr routines are not required as the macros
detect the correct size and type of vector lists.

The number of files changed is fair size, but most of the changes are minor
renaming of macros.

Signed-off-by: Keith Wiles <keith.wiles@intel.com>

### CODE REVIEW

File: test/testcne/vec_test.c [hunk 1]
```diff
@@ -121,12 +158,36 @@ vec_main(int argc, char **argv)
...
+
+        if (ret == TST_FAILED)
+            break;
+    }
 
-    for (int i = 0; i < cne_countof(tdata); i++)
-        test1(&tdata[i]);
-    for (int i = 0; i < cne_countof(tdata); i++)
-        test2(&tdata[i]);
+    tst_end(tst, TST_PASSED);
```
Comment (jeffreybshaw) [inline test/testcne/vec_test.c:170]: Should you mark the test passed even if it failed?

File: lib/include/cne_vec.h [hunk 2]
```diff
@@ -76,66 +56,103 @@ vec_data(void *h)
...
  *
- * @param hv
- *   The vector structure header address
- * @param n
- *   The number of entries in the vector
- * @param flags
- *   The flags to be set in the vector structure
+ * @param vec
+ *   The vector pointer
+ * @param elem
```
Comment (jeffreybshaw) [inline lib/include/cne_vec.h:118]: param name is "nelem". also missing the "esize" param documentation

**Review by jeffreybshaw — CHANGES_REQUESTED** [review]
Please look at the minor doxygen param fix, and a testcne change so that it reports failure when a test fails.

File: lib/cnet/chnl/cnet_chnl.c [hunk 3]
```diff
@@ -56,11 +56,11 @@ _chnl_buf_init(mempool_t *mp, void *opaque_arg __cne_unused, void *_b, unsigned
     if (pthread_cond_init(&ch->ch_rcv.cb_cond, NULL))
         CNE_RET("cond init(ch_rcv) failed\n");
 
-    ch->ch_rcv.cb_vec = vec_alloc_ptr(ch->ch_rcv.cb_vec, CHNL_VEC_SIZE * 4);
+    ch->ch_rcv.cb_vec = vec_alloc(ch->ch_rcv.cb_vec, CHNL_VEC_SIZE);
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply lib/cnet/chnl/cnet_chnl.c:59]: Now that vec code is dynamic and can grow we do not need to preallocate that many ahead of time.

File: lib/cnet/chnl/cnet_chnl.h [hunk 4]
```diff
@@ -38,7 +38,7 @@ struct stk_s;
 extern "C" {
 #endif
 
-#define CHNL_VEC_SIZE 256
+#define CHNL_VEC_SIZE 32
```
[the remark this answers is not in the document]
Comment (KeithWiles) [reply lib/cnet/chnl/cnet_chnl.h:41]: See above

**Review by jeffreybshaw — APPROVED** [review]

**Comment by KeithWiles** [comment]
reverify

**Pull request merged**